### PR TITLE
Switch to setup.cfg; update requirements

### LINKED ISSUE #20: Fix how tests are provided when LOMAP is installed
I'm working on migrating LOMAP to `setup.cfg` (and moving to pip-installability). However, the current `setup.py` has some strange behavior that I think needs to change.

### What LOMAP does now

Currently, `setup` receives `packages=find_packages() + ['test']`. This means that you get 2 packages installed into your `site-packages`: `lomap` and `test`. This copies the test files into `site-packages/test/`, which:

1. Is a really bad usage of namespace, since the contents of `site-packages/test/` [will be whichever package last tried to install something as a package called `test`](https://stackoverflow.com/a/65904451/).
2. Is unusable anyway, since it is shadowed by [standard library `test` package](https://docs.python.org/3/library/test.html), which you'll get on `import test` (unless you did something freaky with your environment).

### Option 1: Make tests a subpackage

Move `test/` -> `lomap/tests/`. It gets installed as usual. Things can be imported from `lomap.tests`. This is what we do with all our other packages.

### Option 2: Include tests via MANIFEST and package-data

This leaves the files where they are currently, but should make tests available at `lomap.tests`. Honestly, this isn't an approach I use, and my attempts to get things like this to work in the past were a bit of a headache.

### Option 3: Don't include tests

We could just say that the tests have to be downloaded separately. I'm not a fan of this.

---

Personally, I'd vote for option 1. I don't think this counts as an API break (technically, we're moving tests to a different namespace, but since it was impossible to access tests with a default Python setup, I don't think it is a problem.)

### PR DESCRIPTION
**Pull request by dwhswenson**
Plans for this PR:

- [x] Switch to a `setup.cfg` based approach
- [x] Check that it installs in an empty environment
- [x] Move tests into the package (Resolves #20)
- [x] Check if there's an rdkit minimum version we need to mark
- [x] Update CI to include Python 3.9, 3.10
- ~Switch CI to use pip installs?~

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,3 +1,52 @@
...
+long_description_type = text/markdown
+author = Gaetano Calabro and David Mobley
+author_email = 
+maintainer = Open Free Energy
+maintainer_email = openfreeenergy@omsf.io
+license = MIT
+license_file = LICENSE.txt
+url = https://github.com/OpenFreeEnergy/Lomap
+classifiers =
+    Development Status :: 3 - Alpha
```
Comment (dwhswenson) [inline setup.cfg:14]: Shall we move it up to at least beta? Seems in moderately wide use now....

File: setup.cfg [hunk 2]
```diff
@@ -1,3 +1,52 @@
+[metadata]
+name = lomap2
+description = Lead Optimization Mapper 2
+long_description = file: README.md
+long_description_type = text/markdown
+author = Gaetano Calabro and David Mobley
+author_email =
```
Comment (dwhswenson) [inline setup.cfg:7]: leaving this blank; previously had `gcalabro -at- uci.edu`: I think that this are supposed to actually be clickable for PyPI, and I wouldn't be sure that the UCI email is still alive.

File: setup.cfg [hunk 3]
```diff
@@ -1,3 +1,52 @@
+[metadata]
+name = lomap2
+description = Lead Optimization Mapper 2
+long_description = file: README.md
+long_description_type = text/markdown
+author = Gaetano Calabro and David Mobley
```
Comment (dwhswenson) [inline setup.cfg:6]: Add Toni here? She had been listed as a maintainer previously, but now we've taken that.

File: setup.cfg (hunk 1 above)
Comment (richardjgowers) [reply setup.cfg:14]: it's 2.x, it's stable

File: setup.cfg (hunk 3 above)
Comment (richardjgowers) [reply setup.cfg:6]: I think those two are still the authors, then there's a bunch of contributors/maintainers, it's tricky

**Comment by richardjgowers** [comment]
re: rdkit minimum, I think there was an issue with 2018ish iirc, so I'd be tempted to say 2021 or so?

**Comment by dwhswenson** [comment]
FWIW, I made a fresh env, did a `python -m pip install .` (that is, NOT an editable install), and then ran `pytest --pyargs lomap` from my root directory (and `lomap -h`). Everything seems to work -- files get installed correctly, it appears (also checked that it looked right in the sdist).

File: setup.cfg [hunk 4]
```diff
@@ -1,3 +1,53 @@
...
+license_file = LICENSE.txt
+url = https://github.com/OpenFreeEnergy/Lomap
+classifiers =
+    Development Status :: 5 - Production/Stable
+    Environment :: Console
+    Intended Audience :: Science/Research
+    Natural Language :: English
+    Operating System :: MacOS :: MacOS X
+    Operating System :: POSIX :: Linux
+    Programming Language :: Python :: 3.8
```
Comment (IAlibay) [inline setup.cfg:20]: Add 3.9?

We should also add 3.10 to CI at some point.

File: setup.cfg [hunk 5]
```diff
@@ -1,3 +1,53 @@
...
+graphviz =
+    pygraphviz
+test =
+    pytest
+
+[options.entry_points]
+console_scripts =
+    lomap = lomap.dbmol:startup
+
+
```
Comment (IAlibay) [inline setup.cfg:50]:
```suggestion
```
Just nitpicking on empty lines per entry :P

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple of tiny things

File: setup.cfg [hunk 6]
```diff
@@ -1,3 +1,53 @@
+[metadata]
+name = lomap2
```
[the remark this answers is not in the document]
Comment (dwhswenson) [reply setup.cfg:2]: The name here is the name used on PyPI. This must be `lomap2`, as `lomap` is already taken.

**Comment by IAlibay** [comment]
Odd that it's failing on 3.10, there is an rdkit package for it

**Comment by dwhswenson** [comment]
> Odd that it's failing on 3.10, there is an rdkit package for it

Maybe not for RDKit 2021? I'm going to add non-2021 RDKit to CI, and exclude RDKit 2021 / Python 3.10 combo.

**Comment by dwhswenson** [comment]
Leaving discussion of switching from conda install to pip install for CI for another time. The main argument to doing that is that our `setup.cfg` can be the single source of truth on requirements. Release workflow would then be (1) release to PyPI; (2) conda-forge watches PyPI and opens a PR when it sees the new PyPI release.

@IAlibay : I think I addressed your changes!

**Review by IAlibay — APPROVED** [review]
lgtm!

**Pull request merged**